sdk/core/azure-core/src/http/request_activity_policy.cpp

```diff
@@ -23,68 +23,78 @@ std::unique_ptr<RawResponse> RequestActivityPolicy::Send(
     NextHttpPolicy nextPolicy,
     Context const& context) const
 {
-  // Create a tracing span over the HTTP request.
-  std::stringstream ss;
-  // We know that the retry policy MUST be above us in the hierarchy, so ask it for the current
-  // retry count.
-  auto retryCount = RetryPolicy::GetRetryCount(context);
-  if (retryCount == -1)
+  // Find a tracing factory from our context. Note that the factory value is owned by the
+  // context chain so we can manage a raw pointer to the factory.
+  auto tracingFactory = DiagnosticTracingFactory::DiagnosticFactoryFromContext(context);
+  if (tracingFactory)
   {
-    // We don't have a RetryPolicy in the policy stack - just assume this is request 0.
-    retryCount = 0;
-  }
-  ss << "HTTP " << request.GetMethod().ToString() << " #" << retryCount;
-  auto contextAndSpan
-      = Azure::Core::Tracing::_internal::DiagnosticTracingFactory::CreateSpanFromContext(
-          ss.str(), SpanKind::Client, context);
-  auto scope = std::move(contextAndSpan.second);
+    // Create a tracing span over the HTTP request.
+    std::stringstream ss;
+    ss << "HTTP " << request.GetMethod().ToString();
 
-  scope.AddAttribute(TracingAttributes::HttpMethod.ToString(), request.GetMethod().ToString());
-  scope.AddAttribute("http.url", m_inputSanitizer.SanitizeUrl(request.GetUrl()).GetAbsoluteUrl());
-  {
+    CreateSpanOptions createOptions;
+    createOptions.Kind = SpanKind::Client;
+    createOptions.Attributes = tracingFactory->CreateAttributeSet();
```

Comment:
This Factory looks more like a manager or a client.  A factory would typically provide static methods and it is just a class (usually a singleton). 
I know this is not a guideline or rule, but consider a different name for this `tracingFactory` maybe ?

Reply:
The class is a factory class for Span objects (more specifically for ContextAndSpan objects). The name (`DiagnosticTracingFactory`) comes [from the C# DiagnosticScopeFactory class](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/DiagnosticScopeFactory.cs). I can rename it to `ContextAndSpanFactory` if that is more consistent.